src/pages/Lend/LendByCategoryPage.vue

```diff
@@ -557,6 +520,101 @@ export default {
 				this.$kvTrackEvent('Lending', experimentId, version);
 			}
 		},
+		createViewportObserver() {
+			// Watch for this element being in the viewport
+			this.viewportObserver = createIntersectionObserver({
+				targets: [this.$refs.bottom],
+				callback: entries => {
+					entries.forEach(entry => {
+						if (entry.isIntersecting) {
+							// This element is in the viewport, so load the data.\
+							this.fetchLoanData();
+							// this.loadData();
+						}
+					});
+				}
+			});
+			if (!this.viewportObserver) {
+				// Observer was not created, so call loadData right away as a fallback.
+				Promise.all([
+					this.fetchRemainingLoanChannels(),
+					this.fetchRecommendedLoans(20)
+				]).then(() => {
+					this.rowLazyLoadComplete = true;
+					this.activateWatchers();
+
+					const pageViewTrackData = this.assemblePageViewData(this.categories);
+					this.$kvTrackSelfDescribingEvent(pageViewTrackData);
+				});
+			}
+		},
+		destroyViewportObserver() {
+			if (this.viewportObserver) {
+				this.viewportObserver.disconnect();
+			}
+		},
+		fetchLoanData() {
+			const category = this.fetchCategoryIds.shift();
+			if (this.fetchCategoryIds.length) {
+				this.rowLazyLoadComplete = false;
+				// eslint-disable-next-line no-underscore-dangle
+				if (category.__typename === 'RecLoanChannel') {
+					try {
+						const variables = {
+							ids: [category.id],
+							imgDefaultSize: this.showHoverLoanCards ? 'w480h300' : 'w480h360',
+							imgRetinaSize: this.showHoverLoanCards ? 'w960h600' : 'w960h720',
+						};
+						return this.apollo.query({
+							query: recommendedLoansQuery,
+							variables,
+						}).then(({ data }) => {
+							const allRecLoanChannels = data.ml?.getOrderedChannelsByIds ?? [];
+
+							const allRecLoans = allRecLoanChannels.map(channel => {
+								const channelLoans = channel.loans?.values ?? [];
+								if (channelLoans.length) {
+									return {
+										values: this.filterFundedLoans(channelLoans),
+										id: channel.id,
+										name: channel.name,
+										description: channel.description,
+									};
+								}
+								return [];
+							});
+							this.recommendedLoans = allRecLoans;
+							this.rowLazyLoadComplete = true;
+						});
+					} catch (e) {
+						logReadQueryError(e, 'LendByCategory recommendedLoansQuery');
+					}
+				} else {
+					try {
+						return this.apollo.watchQuery({
```

Comment:
This might make more sense as a regular apollo.query instead of apollo.watchQuery.